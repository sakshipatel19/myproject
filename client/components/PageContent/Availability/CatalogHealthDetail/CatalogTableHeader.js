export const tableHeader = [
	{
		name: "Product",
		key: "productName",
		isSortable: true,
		style: { width: "38%" }
	},
	{
		name: "Category",
		key: "subCategory",
		isSortable: true,
		style: { width: "14%" }
	},
	{
		name: "Score",
		key: "score",
		isSortable: true,
		style: { width: "10%" }
	},
	{
		name: "Assortment Index",
		key: "assortmentIndex",
		isSortable: true,
		style: { width: "14%" }
	},
	{
		name: "In-Stock Rate",
		key: "instockRate",
		isSortable: true,
		style: { width: "12%" }
	},
	{
		name: "Buy Box Win",
		key: "buyBoxWin",
		isSortable: true,
		style: { width: "12%" }
	}
];
