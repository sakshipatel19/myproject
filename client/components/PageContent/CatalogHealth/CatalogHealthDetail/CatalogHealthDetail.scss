.summary-title-back-button {
	margin-right: 25px;
}
.catalog-health-detail-container {
	width: 100%;
	.aligncenter {
		color: #212129;
		font-family: "NotoSans-Regular";
		font-size: 14px;
		letter-spacing: 0;
		line-height: 20px;
		.table-cell-product-code {
			font-family: "NotoSans-SemiBold";
		}
		.catalog-detail-score-value {
			display: none;
		}
		&:hover {
			.dot-container,
			.icon-class {
				display: none;
			}
			.catalog-detail-score-value {
				display: inline;
			}
		}
	}
	td,
	th {
		text-align: center;
	}
	th {
		padding-right: 0;
	}
	td:first-child,
	td:nth-child(2),
	td:nth-child(3),
	th:nth-child(1),
	th:nth-child(2),
	th:nth-child(3),
	.table-cell-product-name {
		text-align: left;
	}
}
.catalog-health-search-container {
	margin-bottom: 20px;
	display: flex;
	.catalog-health-search {
		width: 86%;
	}
	.catalog-health-filter {
		display: flex;
		width: 10%;
		height: 18px;
		margin: 6px 0 6px auto;
		padding-left: 3%;
		border-left: 1px solid #e7e7e7;
		color: #212129;
		font-family: "NotoSans-SemiBold";
		font-size: 12px;
		letter-spacing: 1px;
		line-height: 18px;
		margin-bottom: 5px;
		cursor: pointer;
		.icon-class {
			margin-left: 3%;
			margin-right: 10%;
		}
	}
}
.catalog-health-tab-container {
	display: flex;
	margin-bottom: 21px;
	width: 100%;
	.catalog-health-tab {
		.tab-details:last-child {
			margin-right: 0;
		}
	}
	.catalog-health-tab-dropdown {
		display: flex;
		flex-direction: row;
		justify-content: flex-end;
		align-items: center;
		align-content: center;
		margin-left: auto;
		margin-bottom: 20px;
		.dropdown-container {
			width: 160px;
		}
	}
	.download-icon {
		margin-left: 4%;
		margin-top: 8px;
	}
}
