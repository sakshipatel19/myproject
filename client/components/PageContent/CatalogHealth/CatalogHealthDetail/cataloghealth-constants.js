import {
	createActions,
	createActionConstants
} from "../../../../constants/actions";

export const SET_CATALOGUE_HEALTH_TABLE_FILTER =
	"SET_CATALOGUE_HEALTH_TABLE_FILTER";
export const SET_CATALOGUE_HEALTH_SELECTED_TAB =
	"SET_CATALOGUE_HEALTH_SELECTED_TAB";
export const SET_CATALOGUE_HEALTH_SEARCH_TERMS =
	"SET_CATALOGUE_HEALTH_SEARCH_TERMS";

export const FETCH_CATEGORIES_LIST = createActionConstants(
	"FETCH_CATEGORIES_LIST"
);

export const fetchCategoriesList = createActions(FETCH_CATEGORIES_LIST);
