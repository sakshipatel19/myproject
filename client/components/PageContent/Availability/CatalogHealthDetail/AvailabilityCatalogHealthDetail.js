import React, { Component, Fragment } from "react";
import CatalogHealthDetail from "../../CatalogHealth/CatalogHealthDetail";
import { defaultTableFilter } from "../../CatalogHealth/CatalogHealthDetail/cataloghealth-reducers";
import Dot from "../../../common/DotWithLabel";
import { CATALOG_HEALTH_DETAIL_COLORPICKER } from "../../../../constants/colors";
import { apiCallBasedOnRoute } from "../../../../services/apiCallBasedOnRoute";
import { tableHeader } from "./CatalogTableHeader";
import Icon from "../../../common/Icon";
import { AVAILABILITY } from "../../../../constants/PageHeaders";

class AvailabilityCatalogHealthDetail extends Component {
	componentDidMount() {
		let payload = {
			...this.props,
			tableFilter: defaultTableFilter,
			selectedTab: this.props.selectedTab,
			searchTerms: []
		};
		this.props.setSearchTermsFilter([]);
		this.props.setTableFilter(defaultTableFilter);
		apiCallBasedOnRoute(payload);
		this.props.fetchCategoriesList(this.props);
	}
	fetchCatalogueHealthTable = payload => {
		const updatedPayload = {
			...this.props,
			...payload
		};
		this.props.fetchAvailabilityCatalogueHealthTable(updatedPayload);
	};
	fetchCatalogueHealthTabs = payload => {
		const updatedPayload = {
			...this.props,
			...payload
		};
		this.props.fetchAvailabilityCatalogueHealthTabs(updatedPayload);
	};
	setSearchTermsFilter = searchList => {
		this.props.setSearchTermsFilter(searchList);
	};
	setSelectedTab = tab => {
		this.props.setSelectedTab(tab);
	};
	setTableFilter = tableFilter => {
		this.props.setTableFilter(tableFilter);
	};
	onCatalogDetailRetryClick = () => {
		let payload = {
			...this.props,
			tableFilter: defaultTableFilter,
			selectedTab: this.props.selectedTab,
			searchTerms: []
		};
		apiCallBasedOnRoute(payload);
		this.props.fetchCategoriesList(this.props);
	};
	DataRowComponent = row => {
		const { catalogHealthDetails } = this.props?.pageConfig[AVAILABILITY];
		const tableHeader = catalogHealthDetails?.tableHeader;
		return (
			<tr className="aligncenter">
				{tableHeader?.map((col, i) => {

					switch (col.key) {
						case 'productName':
							return (<td className={"table-cell-product-name"} key={i}>
								<div>{row.productName}</div>
								<div className={"table-cell-product-code"}>{row.productId}</div>
							</td>);
							break;
						case 'subCategory':
							return (<td key={i}>{`${row.subCategory}`}</td>);
							break;
						case 'score':
							return (<td className={"table-cell-score"} key={i}>{`${row.score}%`}</td>);
							break;
						default:
							if (col.subHeader && col.subHeader.length > 0) {
								const subColumns = col.subHeader.map((sub, index) => {
									return (<td key={index}>{this.tableCellDotCode(row[sub.key])}</td>)
								});
								return subColumns;
							} else {

								return (<td key={i}>{this.tableCellDotCode(row[col.key])}</td>)
							}
							break;

					}
				})
				}
			</tr>
		);
	};

	tableCellDotCode = val => {
		return (
			<Fragment>
				{val?.scoreStatus !== null ? (
					<Dot
						color={CATALOG_HEALTH_DETAIL_COLORPICKER[val?.scoreStatus]}
						size={12}
					/>
				) : (
						<Icon name="noDataOutline" size={12} />
					)}
				<span
					className="catalog-detail-score-value"
					style={{ color: CATALOG_HEALTH_DETAIL_COLORPICKER[val?.scoreStatus] }}
				>
					{val?.score !== null ? `${val?.score}%` : "NA"}
				</span>
			</Fragment>
		);
	};
	render() {
		const {
			catalogHealthTableData,
			catalogHealthTabData,
			selectedTab,
			tableFilter,
			searchTerms,
			countryCode,
			pageConfig
		} = this.props;
		const { enable, tableHeader } = pageConfig[AVAILABILITY]?.catalogHealthDetails;
		const data =
			catalogHealthTableData.data?.availbilityCatalogTable
				?.availabilityCatalogList || [];
		const paginator = catalogHealthTableData.data?.paginator;
		return (enable ?
			<div>
				<CatalogHealthDetail
					data={data}
					catalogHealthTableData={catalogHealthTableData}
					tabData={catalogHealthTabData}
					selectedTab={selectedTab}
					tableFilter={tableFilter}
					searchTerms={searchTerms}
					countryCode={countryCode}
					paginator={paginator}
					setSearchTermsFilter={this.setSearchTermsFilter}
					setSelectedTab={this.setSelectedTab}
					setTableFilter={this.setTableFilter}
					fetchCatalogueHealthTable={this.fetchCatalogueHealthTable}
					fetchCatalogueHealthTabs={this.fetchCatalogueHealthTabs}
					tableHeader={tableHeader}
					DataRowComponent={this.DataRowComponent}
					onRetryClick={this.onCatalogDetailRetryClick}
				/>
			</div> : null
		);
	}
}
export default AvailabilityCatalogHealthDetail;
