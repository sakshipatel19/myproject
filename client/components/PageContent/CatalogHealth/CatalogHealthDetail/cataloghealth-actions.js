import * as constants from "./cataloghealth-constants";

export const setTableFilter = tableFilter => {
	return {
		type: constants.SET_CATALOGUE_HEALTH_TABLE_FILTER,
		response: tableFilter
	};
};

export const setSearchTermsFilter = searchTerms => {
	return {
		type: constants.SET_CATALOGUE_HEALTH_SEARCH_TERMS,
		response: searchTerms
	};
};

export const setSelectedTab = tab => {
	return {
		type: constants.SET_CATALOGUE_HEALTH_SELECTED_TAB,
		response: tab
	};
};
