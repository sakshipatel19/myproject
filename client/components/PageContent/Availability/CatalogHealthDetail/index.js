import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { fetchAvailabilityCatalogueHealthTabs,fetchAvailabilityCatalogueHealthTable }  from "../availability-actions";
import * as catalogActions from "../../CatalogHealth/CatalogHealthDetail/cataloghealth-actions";
import { fetchCategoriesList } from "../../FiltersBar/filters-bar-actions";
import AvailabilityCatalogHealthDetail from "./AvailabilityCatalogHealthDetail";


const mapStateToProps = store => {
	return {
		...store.availability,
		...store.global.config,
		...store.catalogDetail
	};
};

const mapDispatchToProps = dispatch => {
	return bindActionCreators({ 
		fetchAvailabilityCatalogueHealthTabs,
		fetchAvailabilityCatalogueHealthTable,
		fetchCategoriesList,
		...catalogActions
	}, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(AvailabilityCatalogHealthDetail);
