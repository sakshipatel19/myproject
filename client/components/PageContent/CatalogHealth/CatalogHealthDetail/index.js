import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import * as actions from "./cataloghealth-actions";
import { setSelectedPageInConfig } from "../../../global-actions";
import CatalogHealthDetail from "./CatalogHealthDeatil";

const mapStateToProps = (store, props) => {
	return {
		...props,
		...store.catalogDetail,
		categoryNames: store.global.config.categoryNames
	};
};

const mapDispatchToProps = dispatch => {
	return bindActionCreators({ ...actions, setSelectedPageInConfig }, dispatch);
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(CatalogHealthDetail);
