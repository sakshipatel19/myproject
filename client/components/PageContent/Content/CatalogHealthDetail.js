import { connect } from "react-redux";
import React, { Fragment, Component } from "react";
import { bindActionCreators } from "redux";
import {
	fetchContentCatalogueHealthTabs,
	fetchContentCatalogueHealthTable,
	exportContentCatalogHealthTable
} from "./content-actions";
import * as catalogActions from "../CatalogHealth/CatalogHealthDetail/cataloghealth-actions";
import CatalogHealthDetail from "../CatalogHealth/CatalogHealthDetail";
import { defaultTableFilter } from "../CatalogHealth/CatalogHealthDetail/cataloghealth-reducers";
import Dot from "../../common/DotWithLabel";
import { CATALOG_HEALTH_DETAIL_COLORPICKER } from "../../../../client/constants/colors";
import { fetchCategoriesList } from "../FiltersBar/filters-bar-actions";
import { apiCallBasedOnRoute } from "../../../services/apiCallBasedOnRoute";
import Icon from "../../common/Icon";
import { CONTENT } from "../../../constants/PageHeaders";

const tableHeader = [
	[
		{
			name: "Name",
			key: "productName",
			isSortable: true,
			rowSpan: 2,
			style: { width: "28.16%" }
		},
		{
			name: "Category",
			key: "subCategory",
			isSortable: true,
			rowSpan: 2,
			style: { width: "10.94%" }
		},
		{
			name: "Score",
			key: "score",
			isSortable: true,
			rowSpan: 2,
			style: { width: "5.2%" }
		},
		{
			name: "Title",
			key: "title",
			isSortable: true,
			rowSpan: 2,
			style: { width: "6.93%" }
		},
		{
			name: "Image",
			key: "image",
			colSpan: 2,
			style: { width: "12.35%", borderBottomColor: "#858585" }
		},
		{
			name: "Ratings",
			key: "ratings",
			isSortable: true,
			rowSpan: 2,
			style: { width: "6.93%" }
		},
		{
			name: "Reviews",
			key: "reviews",
			isSortable: true,
			rowSpan: 2,
			style: { width: "6.71%" }
		},
		{
			name: "Features & Benefits",
			key: "featuresBullets",
			isSortable: true,
			rowSpan: 2,
			style: { width: "7.58%" }
		},
		{
			name: "Amazon Pref Tags",
			key: "amznPrefTags",
			isSortable: true,
			rowSpan: 2,
			style: { width: "6.93%" }
		},
		{
			name: "Product Desc",
			key: "productDesc",
			isSortable: true,
			rowSpan: 2,
			style: { width: "6.71%" }
		}
	],
	[
		{
			name: "Pri.",
			key: "primaryImage",
			isSortable: true,
			style: { width: "6.7%", paddingLeft: "2%" }
		},
		{
			name: "Sec.",
			key: "secondaryImage",
			isSortable: true,
			style: { width: "5.8%" }
		}
	]
];

class ContentCatalogHealthDetail extends Component {
	componentDidMount() {
		let payload = {
			...this.props,
			tableFilter: defaultTableFilter,
			selectedTab: this.props.selectedTab,
			searchTerms: []
		};
		this.props.setSearchTermsFilter([]);
		this.props.setTableFilter(defaultTableFilter);
		apiCallBasedOnRoute(payload);
		this.props.fetchCategoriesList(this.props);
	}

	fetchCatalogueHealthTable = payload => {
		const updatedPayload = {
			...this.props,
			...payload
		};
		this.props.fetchContentCatalogueHealthTable(updatedPayload);
	};
	fetchCatalogueHealthTabs = payload => {
		const updatedPayload = {
			...this.props,
			...payload
		};
		this.props.fetchContentCatalogueHealthTabs(updatedPayload);
	};
	setSearchTermsFilter = searchList => {
		this.props.setSearchTermsFilter(searchList);
	};
	setSelectedTab = tab => {
		this.props.setSelectedTab(tab);
	};
	setTableFilter = tableFilter => {
		this.props.setTableFilter(tableFilter);
	};
	onCatalogDetailRetryClick = () => {
		let payload = {
			...this.props,
			tableFilter: defaultTableFilter,
			selectedTab: this.props.selectedTab,
			searchTerms: []
		};
		apiCallBasedOnRoute(payload);
		this.props.fetchCategoriesList(this.props);
	};
	DataRowComponent = row => {
		const { catalogHealthDetails } = this.props?.pageConfig[CONTENT];
		const tableHeader = catalogHealthDetails?.tableHeader;
		return (
			<tr className="aligncenter">
				{tableHeader?.map((col, i) => {

					switch (col.key) {
						case 'productName':
							return (<td className={"table-cell-product-name"} key={i}>
								<div>{row.productName}</div>
								<div className={"table-cell-product-code"}>{row.productId}</div>
							</td>);
							break;
						case 'subCategory':
							return (<td key={i}>{`${row.subCategory}`}</td>);
							break;
						case 'score':
							return (<td className={"table-cell-score"} key={i}>{`${row.score}%`}</td>);
							break;
						default:
							if (col.subHeader && col.subHeader.length > 0) {
								const subColumns = col.subHeader.map((sub, index) => {
									return (<td key={index}>{this.tableCellDotCode(row[sub.key])}</td>)
								});
								return subColumns;
							} else {

								return (<td key={i}>{this.tableCellDotCode(row[col.key])}</td>)
							}
							break;

					}
				})
				}
			</tr>
		);
	};

	tableCellDotCode = val => {
		return (
			<Fragment>
				{val?.scoreStatus !== null ? (
					<Dot
						color={CATALOG_HEALTH_DETAIL_COLORPICKER[val?.scoreStatus]}
						size={12}
					/>
				) : (
						<Icon name="noDataOutline" size={12} />
					)}
				<span
					className="catalog-detail-score-value"
					style={{ color: CATALOG_HEALTH_DETAIL_COLORPICKER[val?.scoreStatus] }}
				>
					{val?.score !== null ? `${val?.score}%` : "NA"}
				</span>
			</Fragment>
		);
	};

	onDownloadCatalogHealthDetails = obj => {
		const payload = {
			...this.props,
			...obj
		};

		this.props.exportContentCatalogHealthTable(payload);
	};

	render() {
		const {
			catalogHealthTableData,
			catalogHealthTabData,
			selectedTab,
			tableFilter,
			searchTerms,
			catalogHelathTableExport,
			pageConfig
		} = this.props;
		const { enable, tableHeader } = pageConfig[CONTENT]?.catalogHealthDetails;
		const data =
			catalogHealthTableData.data?.contentCatalogTable?.contentCatalogList ||
			[];
		const paginator = catalogHealthTableData.data?.paginator;
		return (enable ?
			<div>
				<CatalogHealthDetail
					data={data}
					catalogHealthTableData={catalogHealthTableData}
					tabData={catalogHealthTabData}
					selectedTab={selectedTab}
					tableFilter={tableFilter}
					searchTerms={searchTerms}
					paginator={paginator}
					setSearchTermsFilter={this.setSearchTermsFilter}
					setSelectedTab={this.setSelectedTab}
					setTableFilter={this.setTableFilter}
					fetchCatalogueHealthTable={this.fetchCatalogueHealthTable}
					fetchCatalogueHealthTabs={this.fetchCatalogueHealthTabs}
					tableHeader={tableHeader}
					DataRowComponent={this.DataRowComponent}
					onRetryClick={this.onCatalogDetailRetryClick}
					onDownloadClick={this.onDownloadCatalogHealthDetails}
					tableExportData={catalogHelathTableExport}
				/>
			</div> : null
		);
	}
}

const mapStateToProps = store => {
	return {
		...store.contentScore,
		...store.global.config,
		...store.catalogDetail
	};
};

const mapDispatchToProps = dispatch => {
	return bindActionCreators(
		{
			fetchContentCatalogueHealthTabs,
			fetchContentCatalogueHealthTable,
			...catalogActions,
			fetchCategoriesList,
			exportContentCatalogHealthTable
		},
		dispatch
	);
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(ContentCatalogHealthDetail);
