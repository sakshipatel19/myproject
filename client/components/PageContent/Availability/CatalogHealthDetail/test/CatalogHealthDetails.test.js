import React from "react";
import store from "../../../../../../testUtils/mockStore";
import shallowUntilTarget from "../../../../../../testUtils/shallowUntilTarget";
import CatalogHealthDetailParent from "../index";
import AvailabilityCatalogHealthDetail from "../AvailabilityCatalogHealthDetail";
import CatalogHealthDetail from "../../../CatalogHealth/CatalogHealthDetail";
import renderer from "react-test-renderer";

const props = {
	fetchAvailabilityCatalogueHealthTable: jest.fn().mockReturnValue("default"),
	setSearchTermsFilter: jest.fn(),
	setTableFilter: jest.fn(),
	fetchCategoriesList: jest.fn(),
	productIdList: [],
	location: {
		state: {
			catlogKey: "good"
		}
	}
};

describe("<CatalogHealthDetail />", () => {
	let parrentWrapper, wrapper;
	const initialState = store.getState();
	beforeAll(() => {
		wrapper = shallowUntilTarget(
			<CatalogHealthDetailParent {...props} store={store} />,
			AvailabilityCatalogHealthDetail
		);
	});
	afterAll(() => {
		// parrentWrapper.unmount();
		wrapper.unmount();
	});
	test("CatalogHealthDetail component did render", () => {
		expect(wrapper.find(CatalogHealthDetail).length).toEqual(1);
	});
	test("CatalogHealthDetail invoke fetchCatalogueHealthTable", () => {
		const spy = jest.spyOn(wrapper.instance(), "fetchCatalogueHealthTable");
		wrapper.instance().fetchCatalogueHealthTable();
		expect(spy).toHaveBeenCalled();
	});
	test("CatalogHealthDetail invoke fetchCatalogueHealthTabs", () => {
		const spy = jest.spyOn(wrapper.instance(), "fetchCatalogueHealthTabs");
		wrapper.instance().fetchCatalogueHealthTabs();
		expect(spy).toHaveBeenCalled();
	});
	test("CatalogHealthDetail invoke setSearchTermsFilter", () => {
		const spy = jest.spyOn(wrapper.instance(), "setSearchTermsFilter");
		wrapper.instance().setSearchTermsFilter();
		expect(spy).toHaveBeenCalled();
	});
	test("CatalogHealthDetail invoke setTableFilter", () => {
		const spy = jest.spyOn(wrapper.instance(), "setTableFilter");
		wrapper.instance().setTableFilter();
		expect(spy).toHaveBeenCalled();
	});
	test("CatalogHealthDetail invoke setSelectedTab", () => {
		const spy = jest.spyOn(wrapper.instance(), "setSelectedTab");
		wrapper.instance().setSelectedTab();
		expect(spy).toHaveBeenCalled();
	});
	test("CatalogHealthDetail invoke onCatalogDetailRetryClick", () => {
		const spy = jest.spyOn(wrapper.instance(), "onCatalogDetailRetryClick");
		wrapper.instance().onCatalogDetailRetryClick();
		expect(spy).toHaveBeenCalled();
	});
	test("CatalogHealthDetail invoke fetchCatalogueHealthTable", () => {
		const spy = jest.spyOn(wrapper.instance(), "fetchCatalogueHealthTable");
		wrapper.instance().fetchCatalogueHealthTable();
		expect(spy).toHaveBeenCalled();
	});
	test("CatalogHealthDetail invoke DataRowComponent", () => {
		const spy = jest.spyOn(wrapper.instance(), "DataRowComponent");
		wrapper.instance().DataRowComponent({
			productName: "Advil",
			productId: "B000GGO2QG",
			subCategory: "Medications",
			score: 17,
			assortmentIndex: { scoreStatus: "good", score: 93 },
			instockRate: { scoreStatus: "needsAttention", score: 23 },
			buyBoxWin: { scoreStatus: "needsAttention", score: 1 }
		});
		expect(spy).toHaveBeenCalled();
	});
	test("CatalogHealthDetail invoke DataRowComponent negative case", () => {
		const rowData = {
			productName: "Advil",
			productId: "B000GGO2QG",
			subCategory: "Medications",
			score: 17,
			assortmentIndex: { scoreStatus: "good", score: 93 },
			instockRate: { scoreStatus: "needsAttention", score: 23 },
			buyBoxWin: { scoreStatus: "needsAttention", score: 1 }
		};
		let row = wrapper.instance().DataRowComponent(rowData);
		let rowComp = renderer.create(row).toJSON();
		expect(rowComp).toMatchSnapshot();
		rowData.assortmentIndex = {
			scoreStatus: null,
			score: null
		};
		row = wrapper.instance().DataRowComponent(rowData);
		rowComp = renderer.create(row).toJSON();
		expect(rowComp).toMatchSnapshot();
	});
});
