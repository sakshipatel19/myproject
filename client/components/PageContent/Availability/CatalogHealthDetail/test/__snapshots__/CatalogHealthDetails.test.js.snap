// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatalogHealthDetail /> CatalogHealthDetail invoke DataRowComponent negative case 1`] = `
<tr
  className="aligncenter"
>
  <td
    className="table-cell-product-name"
  >
    <div>
      Advil
    </div>
    <div
      className="table-cell-product-code"
    >
      B000GGO2QG
    </div>
  </td>
  <td>
    Medications
  </td>
  <td
    className="table-cell-score"
  >
    17%
  </td>
  <td>
    <div
      className="dot-container undefined"
    >
      <svg
        height={12}
        width={12}
      >
        <circle
          cx={6}
          cy={6}
          fill="#0FBD8C"
          r={6}
        />
      </svg>
      <span
        className="dot-text"
      />
    </div>
    <span
      className="catalog-detail-score-value"
      style={
        Object {
          "color": "#0FBD8C",
        }
      }
    >
      93%
    </span>
  </td>
  <td>
    <div
      className="dot-container undefined"
    >
      <svg
        height={12}
        width={12}
      >
        <circle
          cx={6}
          cy={6}
          fill="#FF647C"
          r={6}
        />
      </svg>
      <span
        className="dot-text"
      />
    </div>
    <span
      className="catalog-detail-score-value"
      style={
        Object {
          "color": "#FF647C",
        }
      }
    >
      23%
    </span>
  </td>
  <td>
    <div
      className="dot-container undefined"
    >
      <svg
        height={12}
        width={12}
      >
        <circle
          cx={6}
          cy={6}
          fill="#FF647C"
          r={6}
        />
      </svg>
      <span
        className="dot-text"
      />
    </div>
    <span
      className="catalog-detail-score-value"
      style={
        Object {
          "color": "#FF647C",
        }
      }
    >
      1%
    </span>
  </td>
</tr>
`;

exports[`<CatalogHealthDetail /> CatalogHealthDetail invoke DataRowComponent negative case 2`] = `
<tr
  className="aligncenter"
>
  <td
    className="table-cell-product-name"
  >
    <div>
      Advil
    </div>
    <div
      className="table-cell-product-code"
    >
      B000GGO2QG
    </div>
  </td>
  <td>
    Medications
  </td>
  <td
    className="table-cell-score"
  >
    17%
  </td>
  <td>
    <img
      className="icon-class "
      height={12}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src={
        Object {
          "canInstrument": true,
          "createTransformer": [Function],
          "getCacheKey": [Function],
          "process": [Function],
        }
      }
      title=""
      width={12}
    />
    <span
      className="catalog-detail-score-value"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      NA
    </span>
  </td>
  <td>
    <div
      className="dot-container undefined"
    >
      <svg
        height={12}
        width={12}
      >
        <circle
          cx={6}
          cy={6}
          fill="#FF647C"
          r={6}
        />
      </svg>
      <span
        className="dot-text"
      />
    </div>
    <span
      className="catalog-detail-score-value"
      style={
        Object {
          "color": "#FF647C",
        }
      }
    >
      23%
    </span>
  </td>
  <td>
    <div
      className="dot-container undefined"
    >
      <svg
        height={12}
        width={12}
      >
        <circle
          cx={6}
          cy={6}
          fill="#FF647C"
          r={6}
        />
      </svg>
      <span
        className="dot-text"
      />
    </div>
    <span
      className="catalog-detail-score-value"
      style={
        Object {
          "color": "#FF647C",
        }
      }
    >
      1%
    </span>
  </td>
</tr>
`;
